src/main/java/org/prebid/server/bidder/adhese/AdheseBidder.java

```diff
@@ -109,36 +110,35 @@ private ExtImpAdhese parseImpExt(Imp imp) {
         }
     }
 
-    private String buildUrl(BidRequest request, ExtImpAdhese extImpAdhese) {
-        return String.format("%s%s%s%s%s%s",
-                getUrl(extImpAdhese),
-                getSlotParameter(extImpAdhese),
-                getTargetParameters(extImpAdhese),
-                getGdprParameter(request.getUser()),
-                getRefererParameter(request.getSite()),
-                getIfaParameter(request.getDevice()));
+    private AdheseRequestBody buildBody(BidRequest request, ExtImpAdhese extImpAdhese) {
+        final Map<String, List<String>> parameterMap = new TreeMap<>();
```

Comment:
HashMap seems to be better choice

Reply:
We want this sorted so it's deterministic in testing